src/main/scala/com.snowplowanalytics.snowplow.badrows/Payload.scala

```diff
@@ -260,9 +260,38 @@ object Payload {
   }
 }
 
-/** Helper used to define a [[Payload.CollectorPayload]]. Isomoprhic to `org.apache.http.NaveValuePair` */
+/** Helper used to define a `CollectorPayload` and a `RawEvent`. Isomoprhic to `org.apache.http.NameValuePair` */
 final case class NVP(name: String, value: Option[String])
 object NVP {
   implicit val nvpEncoder: Encoder[NVP] = deriveEncoder
   implicit val nvpDecoder: Decoder[NVP] = deriveDecoder
+  implicit val nvpsDecoder: Decoder[List[NVP]] = new Decoder[List[NVP]] {
+    final def apply(cur: HCursor): Decoder.Result[List[NVP]] =
+      cur.focus  match {
+        case Some(json) =>
+          json.fold(
+            DecodingFailure("query string (payload.raw.parameters) can not be null", cur.history).asLeft,
+            b => DecodingFailure(s"query string (payload.raw.parameters) can not be boolean, [$b] provided", cur.history).asLeft,
+            n => DecodingFailure(s"query string (payload.raw.parameters) cannot be number, [$n] provided", cur.history).asLeft,
+            s => DecodingFailure(s"query string (payload.raw.parameters) cannot be string, [$s] provided", cur.history).asLeft,
+            a => arrayToNVPs(a),
+            o => objectToNVPs(o)
+          )
+        case None => DecodingFailure("query string (payload.raw.parameters) is missing", cur.history).asLeft
+      }
+  }
+
+  private def arrayToNVPs(arr: Vector[Json]): Decoder.Result[List[NVP]] = {
+    import cats.implicits._
```

Comment:
Let's make it top-level import. If you want to be granular - you need:

```scala
import cats.instances.either._
import cats.instances.list._
import cats.syntax.traverse._
```

Reply:
Ah yes I was planning to fix this before requesting a review but you were quicker than me.

When putting it at the top level it was conflicting with `.widen` :

```
[error] /home/ben/code/badrows/src/main/scala/com.snowplowanalytics.snowplow.badrows/Payload.scala:42:45: value widen is not a member of io.circe.Decoder[com.snowplowanalytics.snowplow.badrows.Payload.RawPayload]
[error]     RawPayload.payloadRawPayloadJsonDecoder.widen,
```

And when using the 3 lines that you wrote :

```
[error] /home/ben/code/badrows/src/main/scala/com.snowplowanalytics.snowplow.badrows/Payload.scala:289:8: value traverse is not a member of List[io.circe.Json]
[error] possible cause: maybe a semicolon is missing before `value traverse'?
[error]       .traverse(_.as[NVP])`
```

I'll take care of it.

Reply:
The problem was for when there was
```
import cats.syntax.functor._
import cats.implicits._
```
or
```
import cats.syntax.functor._
import cats.syntax.either._
import cats.syntax.list._
import cats.syntax.traverse._
```
Having only
```
import cats.implicits._
```
works